packages/aragon-wrapper/src/identity/LocalLabelAddressProvider.js

```diff
@@ -0,0 +1,67 @@
+import { BehaviorSubject } from 'rxjs/Rx'
+import Cache from '../cache'
+import AddressIdentityProvider from './AddressIdentityProvider'
+
+/**
+ * An local label identity provider for addresses
+ *
+ * @class LocalLabelIdentityProvider
+ */
+export default class LocalLabelIdentityProvider extends AddressIdentityProvider {
+  /**
+   * Create a new local label identity provider attached to a globally-stored cache.
+   *
+   * @param {Object} [target=window.parent] An window implementing the postMessage API.
+   */
+  constructor () {
+    super()
+    this.labelCache = new Cache('localAddressLabel')
+  }
+
+  async init () {
+    await this.labelCache.init()
+
+    this.labels = new BehaviorSubject(await this.labelCache.getAll())
+      .scan((labels, modifier) => modifier(labels))
```

Comment:
Need to check the rxjs on this (do we need a `.connect()` afterwards? Is `BehaviorSubject` needed if we do a `.scan()`?)

Reply:
[BehaviorSubject](https://rxjs.dev/api/index/class/BehaviorSubject) doesn't require a connect because it's a hot observable.

If the goal is for the observable to keep the latest snapshot of all local identities I'll see if the `scan` operator poses any problems.

Reply:
As discussed directly, this should look something like:
```
    this.identities$ = new BehaviorSubject(await this.identityCache.getAll())
      .pipe(
        scan((identities, modifier) => modifier(identities)),
        publishReplay(1)
      )
    this.identities$.connect()
```